services/service-core/src/auth/keycloak.service.ts

```diff
@@ -13,11 +13,15 @@
 import { Injectable } from '@nestjs/common';
 import { verify } from 'jsonwebtoken';
 import axios from 'axios';
+import { ConfigService } from '@nestjs/config';
 
 @Injectable()
 export class KeycloakService {
-  public realmUrl =
-    'https://dev.supply-trail.humanitech.net/auth/realms/humanitech';
+  constructor(private readonly configService: ConfigService) {}
+
+  realmUrl = 'https://dev.supply-trail.humanitech.net/auth/realms/humanitech';
```

Comment:
It would be better to have this somewhere else, either in dedicated constants or even better in the service config. Being placed here this way it is a classic case of a "magic string". Also, it seems that it has to be readonly.